ratel/src/ast/operator.rs

```diff
@@ -121,8 +121,8 @@ impl OperatorKind {
     }
 
     #[inline]
-    pub fn as_str(&self) -> &'static str {
-        match *self {
+    pub fn as_str(self) -> &'static str {
```

Comment:
Doesn't look right to me.

Reply:
What about it looks wrong? There's no reason to use a reference here.

Reply:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#wrong_self_convention

For the getter methods below, they also should take `&self` IMO (convention for getter).

Reply:
If I change it back to `&self`, it says to use `self` to save a dereference. It doesn't complain at all if I use `self`.

Reply:
Yes because it's public and that's allow by default because changing interface is a breaking change. Clippy does have lists that disagree (conflicts), so you need to disable the list about dereference here.

Reply:
I disabled `trivially_copy_pass_by_ref` and now it just accepts either, even if I add `#[cfg_attr(feature = "cargo-clippy", warn(wrong_pub_self_convention))]`. I'll change them back to use `&self`, but without the dereference, as that isn't necessary with the new `match` behavior.